src/main/java/com/google/sps/servlets/StartGameInstanceServlet.java

```diff
@@ -2,24 +2,42 @@
 
 import com.google.gson.Gson;
 import com.google.gson.JsonObject;
+import java.util.List; 
 
 import java.io.IOException;
 
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.FirestoreOptions;
+import com.google.cloud.firestore.DocumentReference;
+import com.google.cloud.firestore.CollectionReference;
+import com.google.cloud.firestore.QueryDocumentSnapshot;
+import com.google.cloud.firestore.QuerySnapshot;
+import com.google.cloud.firestore.Query;
+import com.google.api.core.ApiFuture;
+import com.google.cloud.firestore.DocumentSnapshot;
 
 import com.google.sps.daos.GameInstanceDao;
 import com.google.sps.data.GameInstance;
 
+import java.util.List; 
+import java.util.ArrayList; 
+
 @WebServlet("/startGameInstance")
 public class StartGameInstanceServlet extends HttpServlet {
-  
+
+    private List<QueryDocumentSnapshot> questions = new ArrayList<>();
+    private int indexQuestions;
```

Comment:
I think this index should be a property of GameInstance and not stored in the servlet's state. If the server shuts down and/or two requests are handled by different server instances this index would be incorrect.

Reply:
And furthermore, it should be updated in a transaction to ensure concurrent requests do not read a stale value.

Reply:
Okay you are right